app/src/main/java/com/example/sarahtang/mdbsocials/DetailsActivity.java

```diff
@@ -0,0 +1,220 @@
+        fabStar.setOnClickListener(this);
+
+        imageDetail = (ImageView) findViewById(R.id.imageDetail);
+        nameDetail = (TextView) findViewById(R.id.nameDetail);
+        interestedDetail = (TextView) findViewById(R.id.interestedDetail);
+        hostDetail = (TextView) findViewById(R.id.hostDetail);
+        descriptionDetail = (TextView) findViewById(R.id.descriptionDetail);
+
+        //Getting information from intent from socials adapter
+        Intent intent = getIntent();
+        String name = intent.getStringExtra(SocialsAdapter.NAME_KEY);
+        String host = intent.getStringExtra(SocialsAdapter.HOST_KEY);
+        String description = intent.getStringExtra(SocialsAdapter.DESCRIPTION_KEY);
+        String image = intent.getStringExtra(SocialsAdapter.FIREBASE_URL);
+        final int numberInterested = intent.getIntExtra(SocialsAdapter.NUMBER_INTERESTED, -1);
+        ArrayList<String> resultArray = intent.getStringArrayListExtra(SocialsAdapter.PEOPLE_INTERESTED);
+        if (resultArray == null) {
+            resultArray = new ArrayList<>();
+        }
+        peopleInterested = resultArray;
+        firebaseKey = intent.getStringExtra(SocialsAdapter.FIREBASE_KEY);
+
+        //Download image asynchronously
+        class DownloadFilesTask extends AsyncTask<String, Void, Bitmap> {
+            protected Bitmap doInBackground(String... strings) {
+                try {return Glide.
+                        with(getApplicationContext()).
+                        load(strings[0]).
+                        asBitmap().
+                        into(100, 100). // Width and height
+                        get();}
+                catch (Exception e) {return null;}
+            }
+
+            protected void onProgressUpdate(Void... progress) {}
+
+            protected void onPostExecute(Bitmap result) {
+                imageDetail.setImageBitmap(result);
+            }
+        }
+
+
+        //Retrieving image from firebase storage
+        FirebaseStorage.getInstance().getReferenceFromUrl("gs://mdbsocials-fdfae.appspot.com").child(image + ".png").getDownloadUrl().addOnSuccessListener(new OnSuccessListener<Uri>() {
+            @Override
+            public void onSuccess(Uri uri) {
+                (new DownloadFilesTask()).execute(uri.toString());
+                Log.d("GotImageFromStorage", uri.toString());
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception exception) {
+                Log.d("NoImageFromStorage", exception.toString());
+            }
+        });
+
+        //Setting text for each Text view in layout
+        nameDetail.setText(name);
+        hostDetail.setText(getString(R.string.hostedBy) + host);
+        descriptionDetail.setText(getString(R.string.descriptionIntro) + description);
+
+        switch(numberInterested) {
+            case 1:
```

Comment:
While switch cases are nice, there's really no point if you're doing a simple if else thing. Switch case is more for when you have 3 or more cases. Not really a problem, just don't feel pressured to always use a switch case